### PR TITLE
Allow finding leaf by purpose and change tx input interface

### PR DESCRIPTION
**Pull request by romaniy173**
Issue:
1. Add function to get leaf by purpose
2. Disallow to create SelectedTransactionInputs with group, to make possible to filter out leafs from it

### CODE REVIEW

File: BlocksettleNetworkingLib/TransactionData.cpp [hunk 1]
```diff
@@ -86,25 +87,32 @@ bool TransactionData::setWallet(const std::shared_ptr<bs::sync::Wallet> &wallet
 }
 
 bool TransactionData::setGroup(const std::shared_ptr<bs::sync::hd::Group> &group
-   , uint32_t topBlock, bool resetInputs, const std::function<void()> &cbInputsReset)
+   , uint32_t topBlock, bool excludeLegacy, bool resetInputs, const std::function<void()> &cbInputsReset)
 {
    if (!group) {
       return false;
    }
+   std::vector<std::shared_ptr<bs::sync::Wallet>> wallets;
```
Comment (sergey-chernikov) [inline BlocksettleNetworkingLib/TransactionData.cpp:95]: This should go under `group != group_` condition, I guess

File: BlocksettleNetworkingLib/Wallets/SyncHDGroup.cpp [hunk 2]
```diff
@@ -37,6 +37,17 @@ std::shared_ptr<hd::Leaf> hd::Group::getLeaf(const bs::hd::Path &path) const
    return itLeaf->second;
 }
 
+std::shared_ptr<bs::sync::hd::Leaf> bs::sync::hd::Group::getLeaf(bs::hd::Purpose purpose) const
+{
+   for (auto& leaf : getLeaves()) {
```
Comment (sergey-chernikov) [inline BlocksettleNetworkingLib/Wallets/SyncHDGroup.cpp:42]: const qualifier is advised here

File: BlocksettleNetworkingLib/TransactionData.cpp (hunk 1 above)
Comment (romaniy173) [reply BlocksettleNetworkingLib/TransactionData.cpp:95]: No, I need also it in other condition when else if (resetInputs) {

File: BlocksettleNetworkingLib/Wallets/SyncHDGroup.cpp (hunk 2 above)
Comment (romaniy173) [reply BlocksettleNetworkingLib/Wallets/SyncHDGroup.cpp:42]: fixed, thank you

**Pull request merged**